From 26109472cc6db55c5105f794060f3e8f77f813e3 Mon Sep 17 00:00:00 2001
From: Yi Zhao <yi.zhao@windriver.com>
Date: Mon, 6 Jul 2020 08:52:42 +0800
Subject: [PATCH] policy/modules/services/ftp: fix avc denials for ftpd_t with
 MLS policy

* Make ftpd_t domain trusted to write to network objects within its MLS range
* Make ftpd_t domain trusted to inherit and use file descriptors

Fixes:
avc:  denied  { write } for  pid=302 comm="vsftpd" path="socket:[17966]"
dev="sockfs" ino=17966 scontext=system_u:system_r:
ftpd_t:s15:c0.c1023 tcontext=system_u:system_r:init_t:s0-s15:c0.c1023
tclass=unix_stream_socket permissive=0

avc:  denied  { use } for  pid=267 comm="vsftpd" path="/dev/null"
dev="devtmpfs" ino=10064 scontext=system_u:system_r:ftpd_t:s15:c0.c1023
tcontext=system_u:system_r:init_t:s0-s15:c0.c1023 tclass=fd permissive=0

avc:  denied  { use } for  pid=267 comm="vsftpd" path="socket:[17905]"
dev="sockfs" ino=17905 scontext=system_u:system_r:ftpd_t:s15:c0.c1023
tcontext=system_u:system_r:init_t:s0-s15:c0.c1023 tclass=fd permissive=0

Upstream-Status: Pending

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 policy/modules/services/ftp.te | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/policy/modules/services/ftp.te b/policy/modules/services/ftp.te
index 0d84da71c..1feff637f 100644
--- a/policy/modules/services/ftp.te
+++ b/policy/modules/services/ftp.te
@@ -260,6 +260,9 @@ sysnet_use_ldap(ftpd_t)
 userdom_dontaudit_use_unpriv_user_fds(ftpd_t)
 userdom_dontaudit_search_user_home_dirs(ftpd_t)
 
+mls_net_write_within_range(ftpd_t)
+mls_fd_use_all_levels(ftpd_t)
+
 tunable_policy(`allow_ftpd_anon_write',`
 	miscfiles_manage_public_files(ftpd_t)
 ')
-- 
2.17.1

